src/gluonnlp/vocab/bert.py

```diff
@@ -0,0 +1,223 @@
+                                        padding_token, bos_token, eos_token, reserved_tokens)
+        self._mask_token = mask_token
+        self._sep_token = sep_token
+        self._cls_token = cls_token
+
+    @property
+    def mask_token(self):
+        return self._mask_token
+
+    @property
+    def sep_token(self):
+        return self._sep_token
+
+    @property
+    def cls_token(self):
+        return self._cls_token
+
+    def to_json(self):
+        """Serialize BERTVocab object to json string.
+        This method does not serialize the underlying embedding.
+        """
+        if self._embedding:
+            warnings.warn('Serialization of attached embedding '
+                          'to json is not supported. '
+                          'You may serialize the embedding to a binary format '
+                          'separately using bert_vocab.embedding.serialize')
+        vocab_dict = {}
+        vocab_dict['idx_to_token'] = self._idx_to_token
+        vocab_dict['token_to_idx'] = dict(self._token_to_idx)
+        vocab_dict['reserved_tokens'] = self._reserved_tokens
+        vocab_dict['unknown_token'] = self._unknown_token
+        vocab_dict['padding_token'] = self._padding_token
+        vocab_dict['bos_token'] = self._bos_token
+        vocab_dict['eos_token'] = self._eos_token
+        vocab_dict['mask_token'] = self._mask_token
+        vocab_dict['sep_token'] = self._sep_token
+        vocab_dict['cls_token'] = self._cls_token
+        return json.dumps(vocab_dict)
+
+    @classmethod
+    def from_json(cls, json_str):
+        """Deserialize BERTVocab object from json string.
+        Parameters
+        ----------
+        json_str : str
+            Serialized json string of a BERTVocab object.
+        Returns
+        -------
+        BERTVocab
+        """
+        vocab_dict = json.loads(json_str)
+
+        unknown_token = vocab_dict.get('unknown_token')
+        bert_vocab = cls(unknown_token=unknown_token)
+        bert_vocab._idx_to_token = vocab_dict.get('idx_to_token')
+        bert_vocab._token_to_idx = vocab_dict.get('token_to_idx')
+        if unknown_token:
+            bert_vocab._token_to_idx = DefaultLookupDict(bert_vocab._token_to_idx[unknown_token],
+                                                         bert_vocab._token_to_idx)
+        bert_vocab._reserved_tokens = vocab_dict.get('reserved_tokens')
+        bert_vocab._padding_token = vocab_dict.get('padding_token')
+        bert_vocab._bos_token = vocab_dict.get('bos_token')
+        bert_vocab._eos_token = vocab_dict.get('eos_token')
+        bert_vocab._mask_token = vocab_dict.get('mask_token')
+        bert_vocab._sep_token = vocab_dict.get('sep_token')
+        bert_vocab._cls_token = vocab_dict.get('cls_token')
+
+        return bert_vocab
+
+    @classmethod
+    def from_vocab(cls, vocab):
+        """Create BERTVocab object from Vocab object.
+        Parameters
```

Comment:
newline before Parameters, Returns

Reply:
Why do we need this API? Would from/to json be sufficient? This API is a bit confusing